torch/csrc/dynamo/eval_frame.c

```diff
@@ -436,12 +493,24 @@ static CacheEntry* create_cache_entry(
   // equivalent to CacheEntry(check_fn, code, next) in Python
   PyObject* args = Py_BuildValue("OOO", check_fn, code, next);
   CacheEntry* e = (CacheEntry*)PyObject_CallObject((PyObject*)&CacheEntryType, args); // new reference
-  // CacheEntry e is the now the owner of old cachey entry next. This happens
+
+  // create backward link
+  if ((PyObject*)next != Py_None) {
+    // check if next is a CacheEntry
+    DEBUG_CHECK(PyObject_IsInstance((PyObject*)next, (PyObject*)&CacheEntryType));
+    // borrowed
+    next->prev = e;
+  }
+
+  // CacheEntry e is the now the owner of old cachy entry next. This happens
   // when we incref the next pointer in cache_entry_init.
   Py_DECREF(next);
   Py_DECREF(check_fn);
   Py_DECREF(code);
   Py_DECREF(args);
+
+  e->extra_state = extra_state;
+
```

Comment:
Do we need to set `e->weakreflist = NULL` here?

Reply:
I don't think so, cpython docs say
>The object’s constructor should leave it NULL (which is automatic when using the default [tp_alloc](https://docs.python.org/3.11/c-api/typeobj.html#c.PyTypeObject.tp_alloc)).